### ISSUE TITLE
ikhal freezes when installed from github

### ISSUE THREAD

**Issue by pfertyk** [pimutils/khal#493]
I tried to install khal using this command:
```
pip install git+git://github.com/pimutils/khal.git
```
After that when I check the version (`khal --version`) I get:
```
khal, version 0.8.2.dev231+ng6ed1f1b
```
I can run khal without problems, but ikhal freezes. I have to use Ctrl+C to close the program. 
On the other hand, installing khal using pip (`pip install khal`) works perfectly fine. I get the version:
```
khal, version 0.8.3
```
and both khal and ikhal work.

After a short debugging, I found out that the problem is here:
https://github.com/pimutils/khal/blob/master/khal/ui/__init__.py#L1396
For some reason the loop cannot be started correctly.

I am using Python 3.5.2 on Ubuntu 16.04.1 LTS.

This is a problem for me because I would like to start contributing to khal, but I cannot run it properly in development environment (the problem originally occurred when I tried to run `python khal interactive` from my local repository).

**Comment by geier** [comment]
try cloning and than install with `pip install -e .` (that is much nicer for working on it anyway).

**Comment by pfertyk** [comment]
I tried this:
```
git clone git@github.com:pimutils/khal
cd khal/
pip install --user -e .
```
ikhal still doesn't work as expected. I left the configuration file from the time when I run khal and ikhal installed by pip, but I don't think it matters.

**Comment by geier** [comment]
strange, I cannot reproduce this.
Can you try to either install in a virtual environment and/or checking if any of the old release show the same behaviour? (e.g. `git checkout v0.8.0`) If you are up to it, you could then try to `bisect` the error (if one of the old versions from git work).

**Comment by pfertyk** [comment]
I tried checking out versions 0.8.2, 0.8.0 and 0.7. In the last case I wasn't able to run `python khal interactive` (there was no `__main__.py` file inside `khal` folder). Other versions have the same issue. Since it is very unlikely for those versions to be damaged in some way, I think it's obvious that my problems are caused not by latest commits, but by my setup. I will investigate the issue using virtualenvs and maybe other versions of Xubuntu as soon as I find some time.

**Comment by youngmoney** [comment]
Try
pip3 install -e .
In the repo directory, should install it
(Or just pip if you use or Python 3 by default)

This works in a virtual env too

**Comment by pfertyk** [comment]
@youngmoney as you can read in one of the comments above, I have already tried that, but thanks for a hint anyway.

I created virtualenvs with Python3.3 and Python3.4 (in addition to my default Python3.5). This problem occurred on all of them.

To reiterate, this is how my process looks like:
```
mkvirtualenv -p /usr/bin/python3 khal # or python3.3 or python3.4
git clone git@github.com:pimutils/khal
cd khal
pip install -e .
khal configure # (chosen options: 0, 0, y, 0, y, n, y, y)
ikhal
```
And then `ikhal` freezes.

Before running `pip install` I always confirm that my virtualenv is using Python3 and pip3 by default. And if I install khal not from a repository (using `pip install khal` in another virtualenv) then everything works fine.

**Comment by pfertyk** [comment]
I hope I have found a solution.

I did the steps from the comment above, but this time (before installing) I run `git checkout v0.8.3` (the same version that is displayed when running `khal --version` when khal is installed from pip repo). Both khal and ikhal seem to be working fine. I also confirmed this for version v0.8.2 and v0.8.x.

I have no idea why v0.8.2 didn't work before. There might be an issue with the virtualenv I used then.

But that leaves me with a problem: where should I commit to if I want to fix issues with khal? When I fixed the typo in the documentation, I created a PR directly to `master`, but `master` branch seems not to work. Should I create new branches from `v0.8.x` and make PRs to that branch?

**Comment by geier** [comment]
Can you check if v0.8.0 works and if so do a `git bisect` to the current version and see if which commit introduced the freezing behavior?

For most things doing a PR against v0.8.x should be fine as they should be applicable to master as well. If you tell us which part of khal you want to improve, I can tell you with more certainty if this will lead to conflicts or not.

**Comment by pfertyk** [comment]
I used `git bisect` to pinpoint the incorrect commit. After each checkout I run this command:
```
pip uninstall -y khal && pip install -e .
```
to be sure that the instalation is fresh. This is the result:
```
8990ab89df1036f48b39d741dac20d7b59ed9563 is the first bad commit
commit 8990ab89df1036f48b39d741dac20d7b59ed9563
Author: Christian Geier <geier@lostpackets.de>
Date:   Tue Jul 12 01:48:31 2016 +0200

    Dealing with days with lots of events
    
    If a day has more events than the we have lines available, scroll
    through them.

:040000 040000 5a806e4fb43147a719cf20ba699b75df0ab00d05 376b90322e4d4ef363664dd0f0718d7009309c0c M	khal
```

I confirmed that commit `8990ab8` doesn't work, but the previous one (`dd798e0`) is fine, however the displayed information is a bit different that version 0.8.3 (it contains a lot more text, I will attach a screenshot).

**Comment by pfertyk** [comment]
![khal_lots_of_text](https://cloud.githubusercontent.com/assets/11876747/18286733/6a7ae9a6-7474-11e6-86c5-a1854a1031ab.png)

**Comment by pfertyk** [comment]
Regarding your question @geier I would like to try solving [issue 436](https://github.com/pimutils/khal/issues/436). It is marked as beginner friendly, so it seems like a good place to start.

**Comment by geier** [comment]
Thanks for looking into this, I'll have a look. BTW, what version of urwid are you using?

**Comment by pfertyk** [comment]
I'm using urwid 1.3.1.

**Comment by geier** [comment]
me too, so that can't really be the reason... :-(

**Comment by pfertyk** [comment]
I checked again today (using brand new python3 virtualenv and freshly cloned git repo) and it seems that I can run `ikhal` on `master` without any problems. However, there seems to be a significant difference in the interface. Here is the current official look (`pip install khal`):
![khal_current_interface](https://cloud.githubusercontent.com/assets/11876747/18680753/5173a186-7f65-11e6-8cab-ef46d8ca7d23.png)
and here is what I got when I installed khal from repo (`pip install -e .`):
![khal_new_interface](https://cloud.githubusercontent.com/assets/11876747/18680782/6667bce4-7f65-11e6-9b46-0f88a55e0df3.png)
It seems to me that the official release looks much better, and the master branch displays a lot of rubbish. The output is similar to what I got when I used the last working commit (found using `git bisect`, see the screenshot in one of the previous comments), but it is a bit different.

Is this new look a feature or a bug? If this is how it's supposed to look, then I think we can close this issue. If not, I will try to investigate further.
